tests/bootstrap.js

```diff
@@ -0,0 +1,241 @@
+ *     documentation inline.
+ *
+ */
+'use strict';
+
+(function() {
+  // Values usedd to compute default bootstrap options.
+  const isIe = navigator.userAgent.indexOf('MSIE') !== -1 ||
+      navigator.appVersion.indexOf('Trident/') > -1;
+  const localhosts = ['localhost', '127.0.0.1', '[::1]'];
+  const isLocalhost = localhosts.includes(location.hostname);
+  const isFileUrl = location.origin === 'file://';
+
+  // Default bootstrap options.
+  const options = {
+    // Decide whether to use compmiled mode or not.  Please see issue
+    // #5557 for more information.
+    loadCompressed: isIe || !(isLocalhost || isFileUrl),
+
+    // URL of the blockly repository.  This is needed for a few reasons:
+    //
+    // - We need an absolute path instead of relative path because the
+    //   advanced_playground the and regular playground are in
+    //   different folders.
+    // - We need to get the root directory for blockly because it is
+    //   different for github.io, appspot and local.
+    //
+    // Default value will work so long as top-level page is loaded
+    // from somewhere in tests/.
+    root: window.location.href.replace(/\/tests\/.*$/, '/'),
+
+    // List of deps files to load.  Paths relative to root.
+    depsFiles: [
+      'build/deps.js',
+    ],
+    
+    // List of goog.modules to goog.require.
+    requires: [
+      'Blockly',
+      'Blockly.libraryBlocks',
+      'Blockly.Dart.all',
+      'Blockly.JavaScript.all',
+      'Blockly.Lua.all',
+      'Blockly.PHP.all',
+      'Blockly.Python.all',
+    ],
+
+    // List of scripts to load in compressed mode, instead of
+    // requires.  Paths relative to root.
+    compressedScripts: [
+      'build/blockly_compressed.js',
+      'build/blocks_compressed.js',
+      'build/dart_compressed.js',
+      'build/javascript_compressed.js',
+      'build/lua_compressed.js',
+      'build/php_compressed.js',
+      'build/python_compressed.js',
+    ],
+
+    // Additional scripts to be loaded after Blockly is loaded,
+    // whether Blockly is loaded from compressed or uncompressed.
+    // Paths relative to root.
+    additionalScripts: [
+      'msg/messages.js',
+    ],
+  };
+  if (typeof window.BLOCKLY_BOOTSTRAP_OPTIONS === 'object') {
+    Object.assign(options, window.BLOCKLY_BOOTSTRAP_OPTIONS);
+  }
+
+  /* eslint-disable-next-line no-undef */
+  if (typeof module === 'object' && typeof module.exports === 'object') {
+    // Running in node.js.  Maybe we wish to support this.
+    // blockly_uncompressed formerly did, though it appears that the
+    // code had not been working for some time (at least since PR
+    // #5718 back in December 2021.  For now just throw an error.
+    throw new Error('Bootstrapping in node.js not implemented.');
+  }
+
+  // Create a global variable to remember some stat that will be
+  // needed by later scripts.
+  window.blocklyLoader = {
+    compressed: options.loadCompressed,
+    requires: null,
+    done: null,
```

Comment:
I've added some (purely informational) closure type tags to clarify the types.

I thought about initialising `requires` to `[]`, but I wanted to distinguish between "we are not going to require anything(because we're using compressed mode" and "we are in uncompressed mode but for some reason requiring zero modules".